src/sql/azdata.proposed.d.ts

```diff
@@ -1938,4 +1938,61 @@ declare module 'azdata' {
 			isPrimary: boolean;
 		}
 	}
+
+	export interface ChartData {
+		line: {
+			dataset: number[],
+			datasetLabel: string,
+			backgroundColor?: string;
+		};
+		doughnut: {
+			dataset: number[],
+			labels: string[],
+			colors?: string[];
+		};
+		bar: {
+			dataset: number[],
+			labels: string[],
+			datasetLabel: string,
+			colors?: string | string[];
+		};
+		horizontalBar: {
+			dataset: number[],
+			labels: string[],
+			datasetLabel: string,
+			colors?: string | string[];
+		};
+		pie: {
+			dataset: number[],
+			labels: string[],
+			colors?: string[];
+		};
+		radar: {
+			dataset: number[],
+			datasetLabel: string,
+			backgroundColor?: string;
```

Comment:
I don't think background color should be included in this - that should be based on the current theme for the user. See https://github.com/Microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/dashboard/browser/widgets/insights/views/charts/chartInsight.component.ts#L96

Maybe in the future we can have this be an option - but if we do we should require them to specify light and dark colors to ensure proper color support.

Reply:
the chart data type definition looks weird to me, I was expecting something like this:
```
		export interface lineChartData {
			dataset: number[],
			datasetLabel: string,
			backgroundColor?: string;
		};
		export interface doughnutChartData {
			dataset: number[],
			labels: string[],
			colors?: string[];
		};
```

also, as Charles mentioned, if there are no differences between different chart types, you can use a common data type for all.